From c723189df1642fc60d2a9b73d76e93c5ec477068 Mon Sep 17 00:00:00 2001
From: Hannes Frederic Sowa <hannes@redhat.com>
Date: Wed, 16 Jul 2014 13:11:36 -0400
Subject: [net] ipv6: increase ip6_rt_max_size to 16384

Message-id: <70dfa0741912f9a36b95a500a5ed3e3bf11d395e.1405516049.git.hannes@redhat.com>
Patchwork-id: 85857
O-Subject: [PATCH net RHEL7 2/2] ipv6: increase ip6_rt_max_size to 16384
Bugzilla: 1119364
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1119364
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7697742

This commit has no corresponding upstream commit. A corresponding fix
for upstream had been rejected several times.

We provide this to our customers now, as it is doubtful if we backport
any routing table changes in IPv6 land to this kernel.

As customers seem to hit this more often, increase the default value
slightly to reduce support costs.

Signed-off-by: Hannes Frederic Sowa <hannes@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv6/route.c b/net/ipv6/route.c
index b0c3ed9..07f14a0 100644
--- a/net/ipv6/route.c
+++ b/net/ipv6/route.c
@@ -3082,7 +3082,7 @@ static int __net_init ip6_route_net_init(struct net *net)
 #endif
 
  net->ipv6.sysctl.flush_delay = 0;
- net->ipv6.sysctl.ip6_rt_max_size = 4096;
+ net->ipv6.sysctl.ip6_rt_max_size = 16384;
  net->ipv6.sysctl.ip6_rt_gc_min_interval = HZ / 2;
  net->ipv6.sysctl.ip6_rt_gc_timeout = 60*HZ;
  net->ipv6.sysctl.ip6_rt_gc_interval = 30*HZ;
-- 
1.7.1